ui/packages/app/web/src/store/slices/uiSlice.ts

```diff
@@ -0,0 +1,30 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import type {RootState} from '../index';
+
+// Define a type for the slice state
+interface UiState {
+  darkMode: boolean;
+}
+
+// Define the initial state using that type
+const initialState: UiState = {
+  darkMode: false,
+};
+
+export const uiSlice = createSlice({
+  name: 'ui',
+  initialState,
+  reducers: {
+    // Use the PayloadAction type to declare the contents of `action.payload`
+    setDarkMode: (state, action: PayloadAction<boolean>) => {
+      state.darkMode = action.payload;
```

Comment:
It is so comforting to see that we don't have to worry about the immutability of the state object in the reducer. 😄

Reply:
yessss agreed!